toolchain/sem_ir/typed_insts.h

```diff
@@ -749,6 +749,26 @@ struct FunctionType {
   SpecificId specific_id;
 };
 
+// The type of an associated function within an `impl`, modeled as an underlying
+// `FunctionType` plus the value of the `Self` parameter. This is the type of
+// `(SelfType as Interface).AssociatedFunction`.
+struct FunctionTypeWithSelfType {
+  static constexpr auto Kind =
+      InstKind::FunctionTypeWithSelfType.Define<Parse::NoneNodeId>(
+          {.ir_name = "impl_fn_type",
```

Comment:
```suggestion
          {.ir_name = "fn_type_with_self_type",
```

Not sure if you deliberately didn't change this, but maybe should mirror the type name?